### PR TITLE
feat(cli): add support for self-signed s3 providers for snapshots

### PR DESCRIPTION
**Pull request by DanStough**

### CODE REVIEW

File: pkg/s3/s3.go [hunk 1]
```diff
@@ -90,18 +91,19 @@ func CreateS3BucketUsingAPod(ctx context.Context, clientset kubernetes.Interface
...
-		if err := json.Unmarshal([]byte(line), &createBucketPodOutput); err != nil {
+		if logError = json.Unmarshal([]byte(line), &createBucketPodOutput); logError != nil {
 			continue
 		}
 
 		break
 	}
 
-	if !createBucketPodOutput.Success {
-		return errors.Wrapf(err, "failed to create bucket, please check %s pod logs for more details", createBucketPod.Name)
```
Comment (DanStough) [inline pkg/s3/s3.go:104]: This error was out of scope, so there wouldn't be any failure reported and the pod would stick around

File: deploy/s3-bucket-head.sh [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+fi
+
+TMP_S3_HOST=$(echo "$TMP_S3_ENDPOINT" | awk -F/ '{print $3}')
+export TMP_S3_HOST
+export TMP_S3_S3CMD_FLAGS="--access_key=$TMP_S3_ACCESS_KEY_ID --secret_key=$TMP_S3_SECRET_ACCESS_KEY --host=$TMP_S3_HOST --host-bucket=$TMP_S3_BUCKET_NAME.$TMP_S3_HOST $TMP_S3_S3CMD_SSL_FLAGS"
 
-if s3cmd $TMP_S3_S3CMD_FLAGS ls s3://$TMP_S3_BUCKET_NAME 2>&1 | grep -q 'NoSuchBucket'
+echo "Running with flags: --access_key=***** --secret_key=***** --host=$TMP_S3_HOST --host-bucket=$TMP_S3_BUCKET_NAME.$TMP_S3_HOST $TMP_S3_S3CMD_SSL_FLAGS"
+
+if timeout 5 s3cmd $TMP_S3_S3CMD_FLAGS ls s3://$TMP_S3_BUCKET_NAME
```
Comment (DanStough) [inline deploy/s3-bucket-head.sh:19]: Added the timeout here. If you type in the wrong IP address s3cmd will block and keep retrying.

File: cmd/kots/cli/velero.go [hunk 3]
```diff
@@ -352,6 +353,19 @@ func VeleroConfigureOtherS3Cmd() *cobra.Command {
 				return errors.New("velero namespace not found")
 			}
 
+			var caCertData []byte
+			caCertFile := v.GetString("cacert")
+			if caCertFile != "" {
+				realPath, err := filepath.Abs(caCertFile)
```
Comment (salahalsaleh) [inline cmd/kots/cli/velero.go:359]: why is this needed?

File: deploy/s3-bucket-head.sh (hunk 2 above)
Comment (salahalsaleh) [reply deploy/s3-bucket-head.sh:19]: can we increase the timeout a bit? 5 seconds seem very low

File: pkg/s3/s3.go [hunk 4]
```diff
@@ -90,18 +91,19 @@ func CreateS3BucketUsingAPod(ctx context.Context, clientset kubernetes.Interface
...
+	if !createBucketPodOutput.Success || logError != nil {
```
Comment (salahalsaleh) [inline pkg/s3/s3.go:105]: won't `logError` always be NOT nil? since we're echoing invalid json in the script and unmarshalling these messages will fail?

File: pkg/s3/s3.go [hunk 5]
```diff
@@ -142,18 +144,19 @@ func HeadS3BucketUsingAPod(ctx context.Context, clientset kubernetes.Interface,
...
+		if logError = json.Unmarshal([]byte(line), &headBucketPodOutput); logError != nil {
 			continue
 		}
 
 		break
 	}
 
-	if !headBucketPodOutput.Success {
-		return errors.Wrapf(err, "failed to head bucket, please check %s pod logs for more details", headBucketPod.Name)
+	if !headBucketPodOutput.Success || logError != nil {
```
Comment (salahalsaleh) [inline pkg/s3/s3.go:158]: ditto

File: cmd/kots/cli/velero.go (hunk 3 above)
Comment (DanStough) [reply cmd/kots/cli/velero.go:359]: TBH I got this directly from the Velero CLI. My assumption is that it's because `Abs` also runs [Clean](https://golang.org/pkg/path/filepath/#Clean) which can handle some corner case inputs. I tested and it works without, but I'm inclined to keep unless there are strong objections.

File: deploy/s3-bucket-head.sh (hunk 2 above)
Comment (DanStough) [reply deploy/s3-bucket-head.sh:19]: 15 per our slack thread 👍

File: cmd/kots/cli/velero.go (hunk 3 above)
Comment (salahalsaleh) [reply cmd/kots/cli/velero.go:359]: sounds good

File: pkg/s3/s3.go [hunk 6]
```diff
@@ -142,18 +144,19 @@ func HeadS3BucketUsingAPod(ctx context.Context, clientset kubernetes.Interface,
 	headBucketPodOutput := HeadBucketPodOutput{}
 
 	scanner := bufio.NewScanner(bytes.NewReader(logs))
+	var logError error
```
Comment (salahalsaleh) [inline pkg/s3/s3.go:147]: i believe this var isn't needed anymore, right?

File: pkg/s3/s3.go [hunk 7]
```diff
@@ -90,18 +91,19 @@ func CreateS3BucketUsingAPod(ctx context.Context, clientset kubernetes.Interface
 	createBucketPodOutput := CreateBucketPodOutput{}
 
 	scanner := bufio.NewScanner(bytes.NewReader(logs))
+	var logError error
```
Comment (salahalsaleh) [inline pkg/s3/s3.go:94]: ditto

**Review by salahalsaleh — APPROVED** [review]

**Review by salahalsaleh — APPROVED** [review]
